test-data/unit/stubgen.test

```diff
@@ -758,6 +788,8 @@ x = 1
 class C:
     def g(self): ...
 [out]
+__all__ = ['f', 'x', 'C', 'g']
```

Comment:
Hmm. Maybe it's problematic if stubgen adds `__all__` to the generated stub, even though a symbol in `__all__` at runtime (in this case, `g`) isn't actually defined at runtime. On the other hand, maybe it's outside of stubgen's purview to worry about such things; maybe it should just copy the runtime `__all__` faithfully into the stub even when `__all__` is weird at runtime, as you're doing now.

Reply:
It's probably better to omit the undefined names so the generated stub won't create errors for type checkers, so I pushed that change. Could go either way though.

Reply:
(For future readers of this thread: we decided to go back on this; see the discussion in https://github.com/python/mypy/pull/16356#discussion_r1375333153 for why)